web/docs/guides/with-solidjs.mdx

```diff
@@ -0,0 +1,541 @@
+All we need are the API URL and the `anon` key that you copied [earlier](#get-the-api-keys).
+
+```bash title=".env"
+VITE_SUPABASE_URL=YOUR_SUPABASE_URL
+VITE_SUPABASE_ANON_KEY=YOUR_SUPABASE_ANON_KEY
+```
+
+Now that we have the API credentials in place, let's create a helper file to initialize the Supabase client. These variables will be exposed
+on the browser, and that's completely fine since we have [Row Level Security](/docs/guides/auth#row-level-security) enabled on our Database.
+
+```js title="src/supabaseClient.jsx"
+import { createClient } from '@supabase/supabase-js'
+
+const supabaseUrl = process.env.VITE_SUPABASE_URL
+const supabaseAnonKey = process.env.VITE_SUPABASE_ANON_KEY
+
+export const supabase = createClient(supabaseUrl, supabaseAnonKey)
+```
+
+And one optional step is to update the CSS file `src/index.css` to make the app look nice.
+You can find the full contents of this file [here](https://raw.githubusercontent.com/supabase/supabase/master/examples/react-user-management/src/index.css).
+
+### Set up a Login component
+
+Let's set up a SolidJS component to manage logins and sign ups. We'll use Magic Links, so users can sign in with their email without using passwords.
+
+```jsx title="/src/Auth.jsx"
+import { createSignal } from 'solid-js'
+import { supabase } from './supabaseClient'
+
+export default function Auth() {
+  const [loading, setLoading] = createSignal(false)
+  const [email, setEmail] = createSignal('')
+
+  const handleLogin = async (e) => {
+    e.preventDefault()
+
+    try {
+      setLoading(true)
+      const { error } = await supabase.auth.signIn({ email: email() })
+      if (error) throw error
+      alert('Check your email for the login link!')
+    } catch (error) {
+      alert(error.error_description || error.message)
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  return (
+    <div className="row flex flex-center">
+      <div className="col-6 form-widget" aria-live="polite">
+        <h1 className="header">Supabase + SolidJS</h1>
+        <p className="description">Sign in via magic link with your email below</p>
+        {loading() ? (
+          'Sending magic link...'
+        ) : (
+          <form onSubmit={handleLogin}>
+            <label htmlFor="email">Email</label>
+            <input
+              id="email"
+              className="inputField"
+              type="email"
+              placeholder="Your email"
+              value={email()}
+              onChange={(e) => setEmail(e.target.value)}
+            />
+            <button className="button block" aria-live="polite">
+              Send magic link
+            </button>
+          </form>
+        )}
+      </div>
+    </div>
+  )
+}
+```
+
+### Account page
+
+After a user is signed in we can allow them to edit their profile details and manage their account.
+
+Let's create a new component for that called `Account.js`.
+
+```jsx title="src/Account.jsx"
```

Comment:
```suggestion
```jsx title="src/Account.js"
```

Reply:
My initial goal was to make this similar to the React Quickstart with as few changes as possible. In fact, 
I initially tried getting this to work with the ".js" extension, but the tooling expects ".jsx".